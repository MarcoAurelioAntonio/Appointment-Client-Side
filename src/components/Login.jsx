import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { getUserFromAPI, postUserToAPI } from '../redux/usersSlice';

const Login = () => {
  const [inputs, setInputs] = useState({ name: '' });
  const { status, error } = useSelector((store) => store.users);
  const dispatch = useDispatch();
  const history = useNavigate();

  const handleChange = (ev) => {
    setInputs((prevState) => ({
      ...prevState,
      [ev.target.name]: ev.target.value,
    }));
  };
  const handleLoginClick = (ev) => {
    ev.preventDefault();
    dispatch(getUserFromAPI(inputs.name));
  };

  const handleRegisterClick = (ev) => {
    ev.preventDefault();
    dispatch(postUserToAPI(inputs));
  };

  // Redirect if user successfully logged in
  if (status === 'succeed') {
    setTimeout(() => history('/'));
  }

  return (
    <section className="content-login">
      <form onSubmit={handleLoginClick}>
        <div className="imgcontainer">
          <img className="avatar" src="/images/logo.png" alt="Logo" />
        </div>

        <div className="container">
          <label htmlFor="name">
            <input
              className="field"
              type="text"
              placeholder="Enter Username"
              value={inputs.title}
              id="name"
              name="name"
              required
              onChange={handleChange}
              minLength={3}
            />
          </label>
          <section className="msg-section">
            <p className="error">{error && `Username ${error}`}</p>
          </section>
          <button className="lg-btn" type="submit" disabled={status === 'loading' && true}>
            Login
          </button>
          <button
            className="reg-btn"
            type="button"
            onClick={handleRegisterClick}
            disabled={status === 'loading' && true}
          >
            Register
          </button>
        </div>
      </form>
    </section>
  );
};

export default Login;
