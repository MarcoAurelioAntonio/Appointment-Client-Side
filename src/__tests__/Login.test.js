import { render, screen } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import Login from '../components/Login';

it('Should display 2 buttons on the screen', () => {
  const mockStore = configureStore([]);
  const user = {
    id: 68,
    name: 'Teddy',
  };

  const initialState = {
    users: {
      current_user: user,
      status: 'idle',
      loading: false,
      error: null,
    },
  };

  const store = mockStore(initialState);
  render(
    <BrowserRouter>
      <Provider store={store}>
        <Login />
      </Provider>
    </BrowserRouter>,
  );
  const allLinks = screen.getAllByRole('button');

  expect(allLinks).toHaveLength(2);
  expect(store.getState()).toMatchSnapshot();
});
