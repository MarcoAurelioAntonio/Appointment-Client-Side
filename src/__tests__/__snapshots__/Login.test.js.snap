// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should display 3 countries on the screen 1`] = `
Object {
  "users": Object {
    "current_user": Object {
      "id": 68,
      "name": "Teddy",
    },
    "error": null,
    "loading": false,
    "status": "idle",
  },
}
`;
